.circleci/run-examples.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+import os, sys, time, shlex, subprocess, multiprocessing, examples
+
+def cmd(command, redirect=None, timeout=None):
+  print(">", command)
+  try:
+    return subprocess.call(shlex.split(command), stdout=redirect,
+                           stderr=redirect, timeout=timeout)
+  except subprocess.TimeoutExpired:
+    return 'timeout'
+
+def run_example(name):
+  print("Running", name)
+  artifact_path = f"/tmp/artifacts/{name}"
+  start_time = time.time()
+
+  with open(artifact_path, "w") as artifact:
+    cmd(f'./one.sh clean "{name}"', redirect=artifact)
+    build_ret = cmd(f'./one.sh build "{name}"', redirect=artifact)
+    if build_ret == 0:
+      run_ret = cmd(f'./one.sh run "{name}"', redirect=artifact, timeout=600)
+      if run_ret == 0:
+        status = "pass"
+        outcome = f"{name} passed"
+      elif run_ret == 'timeout':
+        status = "fail-time"
+        outcome = f"{name} timed out"
+      else:
+        status = "fail"
+        outcome = f"{name} failed"
+    else:
+      status = "fail"
+      outcome = f"{name} failed to build"
+
+  duration = int(time.time() - start_time)
+  cmd(f'gzip "{artifact_path}"')
+  print(outcome)
+
+  with open(f"/tmp/test_results/{name}.xml", "w") as xml:
+    maybe_fail = "" if status == "pass" else f'<failure message="{status}">{outcome}</failure>'
+    xml.write(f"""
+      <?xml version="1.0" encoding="UTF-8"?>
+      <testsuite name="examples.{connector}">
+        <testcase name="{name}" time="{duration}">
+        {maybe_fail}
+        </testcase>
+      </testsuite>
+    """)
+
+  rank = os.environ["CIRCLE_NODE_INDEX"]
+  with open(f'/tmp/workspace/record/"{name}"') as record:
+    record.write(f'["{status}", "{connector}.{rank}"]')
+
+def run_multiple_examples(examples):
+  multiprocessing.Pool(len(examples)).map(run_example, examples)
+
+connector = sys.argv[1]
+os.environ["REACH_DEBUG"] = "1"
+os.environ["REACH_CONNECTOR_MODE"] = connector
+
+os.chdir("../examples")
+cmd("mkdir -p /tmp/artifacts /tmp/test_results /tmp/workspace/record")
+cmd("../reach devnet --await-background")
+
+# Generate what to pass to stdin for `circleci tests split`
+# The split command takes some lines, then magically splits the lines
+# between runners and prints which lines the current runner is assigned.
+# We pass it exactly the same number of lines as there are parallel runners,
+# so we get exactly one line back per runner.
```

Comment:
It really isn't "magical" and basically what you're doing is...
```
(special, regular) = examples.split_examples_into_jobs(connector)
comb = special <> regular
comb[CIRCE_NODE_INDEX]
```
But I don't think your `special` is a list of lists, like regular is. My advice... change it so `split_examples_into_jobs` returns `[ [ Job ] ]` where the length is `parallel_jobs_for(conn)` and then you always `run_multiple_examples`, but sometimes the list just has one thing